test_pull_requests

```diff
@@ -862,8 +1058,9 @@ popd
       repo_to_pull_request.each do |repo, pull_request|
         pull_request_comments = get_comments(pull_request['number'], repo)
         test_prefix_comment = get_comment_with_prefix(pull_request['number'], repo, settings['test_prefix'], pull_request_comments)
-        if !test_prefix_comment || !(test_prefix_comment['body'] =~ /^#{settings['test_prefix']} Waiting/)
-          being_queued_comment = "#{settings['test_prefix']} Waiting: Determining build queue position"
+        fields = extract_bot_comment_fields(test_prefix_comment['body'], settings) if test_prefix_comment
+        if !test_prefix_comment || !([:wait_in_queue, :wait_queue_pos, :wait_stable_build].include? fields[:state])
```

Comment:
I'm pretty sure the logic - which was valid before - remains unchanged. If you are in this function, and this PR is being processed (`test_prefix_comment` exists), and you're not in any of the 3 waiting states, then enter the "Determining build queue position" state and update the comment accordingly.

@danmcp I'm not crazy about the `include?` either, but I think it's a pretty well-understood idiom, and I don't know if abstracting it behind a method is going to lead to clearer code. I use the same idiom for a different combination of states here: https://github.com/openshift/test-pull-requests/pull/31/files#diff-c3e533b1b4fb4110b28514d90a379965R1691

I've already added a lot of methods and constants with **very** verbose names; if you have some hot suggestions on succinct, distinct method names for this abstraction I'd like to hear it. I suspect we'll need to add more combined checks like this in the future.

Also @danmcp, I did push back on breaking the `:waiting` state into three distinct states, but @stevekuznetsov made a pretty good argument: it fixes a buggy comparison here: https://github.com/openshift/test-pull-requests/pull/31/files#diff-c3e533b1b4fb4110b28514d90a379965R1231